controllers/k8ssandra/cassandra_telemetry_reconciler_test.go

```diff
@@ -0,0 +1,77 @@
+// Some of the tests in this package target functions from pkg/telemetryapi, because they need to be envtests. We prefer to keep envtests in the controller
+// packages.
+package k8ssandra
+
+import (
+	"context"
+
+	"testing"
+
+	testlogr "github.com/go-logr/logr/testing"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	telemetryapi "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/config"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/telemetry"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/test"
+	promapi "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// new_DummyK8ssandraClusterReconciler gives us back a `K8ssandraClusterReconciler` with just the fields that we need to test `reconcileCassandraDCTelemetry()`.
+func newDummyK8ssandraClusterReconciler() K8ssandraClusterReconciler {
+	return K8ssandraClusterReconciler{ReconcilerConfig: &config.ReconcilerConfig{DefaultDelay: interval}}
+}
+
+// Test_CassPrometheusResourcer_UpdateResources_TracksNamespaces tests that the servicemonitor is created in the namespace of the CassandraDC,
+// not the namespace of the k8ssandraCluster.
+func Test_reconcileCassandraDCTelemetry_TracksNamespaces(t *testing.T) {
```

Comment:
We are relying on `envtest` for controller integration tests. We need a new or updated integration test for the K8ssandraCluster controller (see `k8ssandracluster_controller_test.go`) to verify the ServiceMonitor functionality. 

It would be good to have a happy path test which will of course require having the ServiceMonitor CRD installed. We configure CRDs in `testenv.go`. Assuming you already have unit test coverage for creating the ServiceMonitor, it would be probably be sufficient to check that the object was actually created.

It would be nice to also verify that the ServiceMonitor is deleted when telemetry is disabled.

Reply:
Done, thanks for helping me work through some of that too.